sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithListStateSuite.scala

```diff
@@ -301,6 +301,8 @@ class TransformWithListStateSuite extends StreamTest
         CheckNewAnswer(("k5", "v5"), ("k5", "v6")),
         Execute { q =>
           assert(q.lastProgress.stateOperators(0).customMetrics.get("numListStateVars") > 0)
+          assert(q.lastProgress.stateOperators(0).numRowsUpdated === 6)
+          assert(q.lastProgress.stateOperators(0).numRowsRemoved === 1)
```

Comment:
Yeah, seems like my earlier comment is getting at exactly this. It's 1 here, even though it should probably be 6? To the user, the fact that we use merge for lists is an implementation detail.